x-pack/plugins/security_solution/public/common/hooks/use_upgrade_security_packages.ts

```diff
@@ -9,25 +9,50 @@ import { useEffect } from 'react';
 import type { HttpFetchOptions, HttpStart } from '@kbn/core/public';
 import type { BulkInstallPackagesResponse } from '@kbn/fleet-plugin/common';
 import { epmRouteService } from '@kbn/fleet-plugin/common';
-import { useKibana } from '../lib/kibana';
+import type { InstallPackageResponse } from '@kbn/fleet-plugin/common/types';
+import { KibanaServices, useKibana } from '../lib/kibana';
 import { useUserPrivileges } from '../components/user_privileges';
 
 /**
  * Requests that the endpoint and security_detection_engine package be upgraded to the latest version
  *
  * @param http an http client for sending the request
  * @param options an object containing options for the request
+ * @param prebuiltRulesPackageVersion specific version of the prebuilt rules package to install
  */
 const sendUpgradeSecurityPackages = async (
   http: HttpStart,
-  options: HttpFetchOptions = {}
-): Promise<BulkInstallPackagesResponse> => {
-  return http.post<BulkInstallPackagesResponse>(epmRouteService.getBulkInstallPath(), {
-    ...options,
-    body: JSON.stringify({
-      packages: ['endpoint', 'security_detection_engine'],
-    }),
-  });
+  options: HttpFetchOptions = {},
+  prebuiltRulesPackageVersion?: string
+): Promise<Array<PromiseSettledResult<BulkInstallPackagesResponse | InstallPackageResponse>>> => {
+  const packages = ['endpoint', 'security_detection_engine'];
+  const requests: Array<Promise<InstallPackageResponse | BulkInstallPackagesResponse>> = [];
+
+  // If `prebuiltRulesPackageVersion` is provided, try to install that version
+  // Must be done as two separate requests as bulk API doesn't support versions
+  if (prebuiltRulesPackageVersion != null) {
+    packages.splice(packages.indexOf('security_detection_engine'), 1);
+    requests.push(
+      http.post<InstallPackageResponse>(
+        epmRouteService.getInstallPath('security_detection_engine', prebuiltRulesPackageVersion),
+        {
+          ...options,
+        }
+      )
+    );
+  }
+
+  // Note: if `prerelease:true` option is provided, endpoint package will also be installed as prerelease
+  requests.push(
+    http.post<BulkInstallPackagesResponse>(epmRouteService.getBulkInstallPath(), {
+      ...options,
+      body: JSON.stringify({
+        packages,
+      }),
+    })
+  );
+
+  return Promise.allSettled(requests);
```

Comment:
Nice _catch_ 😅 -- I think this is fine for now since existing error handling is ignored here, but we could always add a toast message or something down the line if we want. I think the spirit of the code here is to be quiet and try to upgrade in the background, so don't pester the user unless necessary. 

I'll leave the try/catch as-is though since `await context.services.fleet?.isInitialized();` can still throw, and we can always manually throw if the `sendUpgradeSecurityPackages` Promise.allSettled() return value contains any errors.

Of course we could go back to `Promise.all()` and return `Promise<Array<InstallPackageResponse | BulkInstallPackagesResponse>>`, however I didn't like that as it would immediately reject if one fails, and if making two requests in the case of supplying a `prebuiltRulesPackageVersion`, it seemed prudent to also capture the response of the second request.